enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/Steps.java

```diff
@@ -0,0 +1,662 @@
+        public NotifyDescriptor createInput() {
+            List<NotifyDescriptor.QuickPick.Item> items = new ArrayList<>(types.length);
+            for (String itemType : types) {
+                items.add(new NotifyDescriptor.QuickPick.Item(itemType, itemType));
+            }
+            return new NotifyDescriptor.QuickPick(Bundle.SelectResourceType(), Bundle.SelectResourceType(), items, false);
+        }
+
+        @Override
+        public boolean onlyOneChoice() {
+            return false;
+        }
+
+        @Override
+        public Step getNext() {
+            return new TenancyStep().prepare(null, lookup);
+        }
+
+        @Override
+        public void setValue(String selected) {
+            context.setItemType(selected);
+        }
+
+        @Override
+        public String getValue() {
+            return context.getItemType();
+        }
+
+    }
+
+    /**
+     * The purpose of this step is to select a project to update dependencies.
+     */
+    static class ProjectStep implements Step<OCIItem, Object> {
+
+        private final CompletableFuture<Project[]> projectsFuture;
+        Project[] projects;
+        private Project selectedProject;
+        private OCIItem item;
+
+        ProjectStep(CompletableFuture<Project[]> projectsFuture) {
+            this.projectsFuture = projectsFuture;
+        }
+
+        @Override
+        public Step<OCIItem, Object> prepare(OCIItem item, Lookup lookup) {
+            this.item = item;
+            try {
+                projects = projectsFuture.get();
+            } catch (InterruptedException ex) {
+                Exceptions.printStackTrace(ex);
+            } catch (ExecutionException ex) {
+                Exceptions.printStackTrace(ex);
+            }
+            return this;
+        }
+
+        @Override
+        public NotifyDescriptor createInput() {
+            List<NotifyDescriptor.QuickPick.Item> items = new ArrayList<>(projects.length);
+            for (Project project : projects) {
+                ProjectInformation pi = ProjectUtils.getInformation(project);
+                items.add(new NotifyDescriptor.QuickPick.Item(
+                        pi.getDisplayName(),
+                        project.getProjectDirectory().getName()));
+            }
+            String title = Bundle.SelectProject();
+            if (projects.length == 0) {
+                title = Bundle.NoProjects();
+            }
+            return new NotifyDescriptor.QuickPick(title, title, items, false);
+        }
+
+        @Override
+        public boolean onlyOneChoice() {
+            return projects.length == 1;
+        }
+
+        @Override
+        public Step getNext() {
+            return null;
+        }
+
+        @Override
+        public void setValue(String selected) {
+            for (Project project : projects) {
+                ProjectInformation pi = ProjectUtils.getInformation(project);
+                if (pi.getDisplayName().equals(selected)) {
```

Comment:
Nitpick: there's a very slight chance that the pi.getName() changes through the time: Maven uses some weird lazy-initialized getDisplayName().